src/index.js

```diff
@@ -55,24 +55,28 @@ export default function urlJoin(...parts) {
     .join('/');
 
     // Split the parts into prefix, pathname, and suffix
-    // (scheme://host)(/pathname)(?queryString)
+    // (scheme://host)(/pathnameParts.join('/'))(?queryString)
     const { prefix, pathname, suffix } = parseUrl(partsStr, options);
 
+    const { parts: pathnameParts, hasLeading, hasTrailing } = pathname;
+
+    const { leadingSlash, trailingSlash } = options;
+
     let url = '';
 
     // Start with prefix if not empty (http://google.com)
     if (prefix) {
-        url += prefix + (pathname.parts.length > 0 ? '/' : '');
-    // Otherwise start with the leading slash
-    } else if (options.leadingSlash) {
+        url += prefix + (pathnameParts.length > 0 ? '/' : '');
```

Comment:
I'm really not sure if I should change this line or not, is this the expected behavior?

Reply:
I think it's correct.